src/pusher/PusherClient.js

```diff
@@ -0,0 +1,59 @@
+const Pusher = require('pusher-js')
+const batchAuthorizer = require('./batchAuthorizer')
+
+let pusher = null
+
+class PusherClient {
+
+  static connect({ apiKey, userId, cluster = 'eu' }) {
+    if (pusher) return pusher
+    pusher = new Pusher(apiKey, { cluster, authorizer: batchAuthorizer(userId) })
+    return pusher
+  }
+
+  static registerConnectionEvent(eventName, eventHandler) {
+    if (!pusher) {
+      console.warn('Initialize pusher before trying to register connection events')
+      return
+    }
+    pusher.connection.bind(eventName, eventHandler)
+  }
+
+  static getClient() { return pusher }
+
+  static resetPusherClient() { pusher = null }
```

Comment:
language is swapping between `pusherClient` and `client`

Reply:
fixed